File: src/main/scala/top/scaleda/idea/windows/bottomPanel/ScaledaMessagesWindowFactory.scala
package top.scaleda
package idea.windows.bottomPanel

import idea.ScaledaBundle
import idea.project.io.YmlRootManager
import idea.windows.bottomPanel.console.ConsoleTabManager
import idea.windows.bottomPanel.message.MessageListPanel

import com.intellij.openapi.project.Project
import com.intellij.openapi.util.Disposer
import com.intellij.openapi.wm.{ToolWindow, ToolWindowFactory}

final class ScaledaMessagesWindowFactory extends ToolWindowFactory {
  override def shouldBeAvailable(project: Project): Boolean = {
    YmlRootManager.getInstance(project).getRoots.nonEmpty
  }

  override def createToolWindowContent(
      project: Project,
      toolWindow: ToolWindow
  ): Unit = {
    val service = project.getService(classOf[ConsoleService])
    val tabManager =
      new ConsoleTabManager(project, toolWindow.getContentManager)
    Disposer.register(service, tabManager)

    val messageTab = new MessageListPanel(project)

    tabManager.addPanel(messageTab, ScaledaBundle.message("windows.tool.log.message.title"))
    tabManager.addConsoleTab(ScaledaBundle.message("windows.tool.log.console.title"), switchTo = false)
  }
}

File: src/main/scala/top/scaleda/idea/windows/bottomPanel/ScaledaMessagesWindowFactory.java
package top.scaleda.idea.windows.bottomPanel;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.openapi.wm.ToolWindowFactory;
import org.jetbrains.annotations.NotNull;
import top.scaleda.idea.ScaledaBundle;
import top.scaleda.idea.project.io.YmlRootManager;
import top.scaleda.idea.utils.ScaledaIdeaLogger;
import top.scaleda.idea.windows.bottomPanel.console.ConsoleTabManager;
import top.scaleda.idea.windows.bottomPanel.message.MessageListPanel;
import top.scaleda.idea.windows.bottomPanel.netviewer.NetViewerPanel;

public class ScaledaMessagesWindowFactory implements ToolWindowFactory {
  @Override
  public boolean shouldBeAvailable(@NotNull Project project) {
    return YmlRootManager.getInstance(project).getRoots().nonEmpty();
  }

  @Override
  public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow) {
    ConsoleService service = project.getService(ConsoleService.class);
    ConsoleTabManager tabManager = new ConsoleTabManager(project, toolWindow.getContentManager());
    service.setConsoleTabManager(tabManager);
    MessageListPanel messageTab = new MessageListPanel(project);
    NetViewerPanel netViewerPanel = new NetViewerPanel(project);
    tabManager.addPanel(netViewerPanel, ScaledaBundle.message("windows.tool.log.waveform.title"), false, false);
    tabManager.addPanel(messageTab, ScaledaBundle.message("windows.tool.log.message.title"), false, false);
    tabManager.addConsoleTab(ScaledaIdeaLogger.getId(), ScaledaBundle.message("windows.tool.log.console.title"), false, false);
  }
}
